### ISSUE TITLE
OnPlatform MarkupExtenstion Issues

### ISSUE THREAD

**Issue by ChasakisD** [xamarin/Xamarin.Forms#3920]
### Description

OnPlatform MarkupExtension seems to cannot provide the value of the corresponding type. For example for the property ```IsClippedToBounds``` when using the new OnPlatform, you get 
```
Can not convert True to type 'System.Boolean'
SetValue: Can not convert True to type 'System.Boolean
```
while using the old one is working fine:
```xml
<Setter Property="IsClippedToBounds">
    <OnPlatform x:TypeArguments="x:Boolean">
        <On Platform="Android" Value="True" />
        <On Platform="iOS" Value="False" />
    </OnPlatform>
</Setter>
```
### Steps to Reproduce

1. Add this setter to a layout's style
```xml 
<Setter Property="IsClippedToBounds" Value="{OnPlatform Android=True, iOS=False}" /> 
```
2. When navigating to a page, in the application output you will get the described warning
3. Use the old OnPlatform and any warning occurs

### Expected Behavior

Any warning should occur and it should work along all properties.

### Actual Behavior

The new OnPlatform has no effect to some properties(such as the CornerRadius of Frame), and to others, it prints a warning.

### Basic Information

- Version with issue: 3.2.0.839982
- Last known good version: 3.1.0, before the new OnPlatform
- IDE: VS For Mac 7.6.7 build 49
- Platform Target Frameworks: 
  - Android: Target Framework API 27
- Android Support Library Version: 27.0.2.1

**Comment by StephaneDelcroix** [comment]
@ChasakisD as you're reporting this being broken inside Styles, I think it's a dupe of #3862

**Comment by StephaneDelcroix** [comment]
I added a fix in #3995. if it doesn't fix it for you, please add a sample to this issue, and ask for it to be reopened
